src/NSwag.SwaggerGeneration.AspNetCore/OperationResponseProcessor.cs

```diff
@@ -0,0 +1,222 @@
+
+            if (context.ApiDescription.SupportedResponseTypes.Any())
+            {
+                var responses = new List<OperationResponseModel>();
+                foreach (var apiResponse in context.ApiDescription.SupportedResponseTypes)
+                {
+                    var returnType = apiResponse.Type;
+                    // Attempt to look up a Swagger attribute by type.
+                    var attribute = responseTypeAttributes.FirstOrDefault(a =>
+                    {
+                        if (a.GetType().GetRuntimeProperty("ResponseType") != null)
+                            return ((dynamic)a).ResponseType == returnType;
+                        else if (a.GetType().GetRuntimeProperty("Type") != null)
+                            return ((dynamic)a).Type == returnType;
+                        else
+                            return false;
+                    });
+
+                    var attributeType = attribute?.GetType();
+                    dynamic responseTypeAttribute = attribute;
+
+                    string httpStatusCode;
+                    if (apiResponse.StatusCode == 0 && IsVoidResponse(returnType))
+                        httpStatusCode = GetVoidResponseStatusCode();
+                    else if (apiResponse.TryGetPropertyValue<bool>("IsDefaultResponse"))
+                        httpStatusCode = "default";
+                    else
+                        httpStatusCode = apiResponse.StatusCode.ToString(CultureInfo.InvariantCulture);
+
+                    if (attributeType?.GetRuntimeProperty("HttpStatusCode") != null && responseTypeAttribute.HttpStatusCode != null)
+                        httpStatusCode = responseTypeAttribute.HttpStatusCode.ToString();
+                    else if (attributeType?.GetRuntimeProperty("StatusCode") != null && responseTypeAttribute.StatusCode != null)
+                        httpStatusCode = responseTypeAttribute.StatusCode.ToString();
+
+                    var description = HttpUtilities.IsSuccessStatusCode(httpStatusCode) ? successXmlDescription : string.Empty;
+                    if (attributeType?.GetRuntimeProperty("Description") != null)
+                    {
+                        if (!string.IsNullOrEmpty(responseTypeAttribute.Description))
+                            description = responseTypeAttribute.Description;
+                    }
+
+                    var isNullable = true;
+                    if (attributeType?.GetRuntimeProperty("IsNullable") != null)
+                        isNullable = responseTypeAttribute.IsNullable;
+
+                    responses.Add(new OperationResponseModel(httpStatusCode, returnType, isNullable, description));
+                }
+
+                foreach (var statusCodeGroup in responses.GroupBy(r => r.HttpStatusCode))
+                {
+                    var httpStatusCode = statusCodeGroup.Key;
+                    var returnType = statusCodeGroup.Select(r => r.ResponseType).FindCommonBaseType();
```

Comment:
`ApiDescription` doesn't really support this - multiple response types with the same status code. We should consider throwing in MVC if someone tries to, but we won't generate a set of actions like this today.